nipype/interfaces/io.py

```diff
@@ -2863,3 +2864,33 @@ def _list_outputs(self):
 
     def _add_output_traits(self, base):
         return add_traits(base, list(self.inputs.output_query.keys()))
+
+
+class ExportFileInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, desc='Input file name')
+    out_file = File(exists=False, desc='Output file name')
+    check_extension = traits.Bool(False, desc='Ensure that the input and output file extensions match')
```

Comment:
Somewhat your call, but I would probably set the default for this to `True`, and require users to turn it off to show they understand that the extension is changing. Otherwise you might get situations where a `.nii` file is copied to ` .nii.gz` file, and most software will get confused.